coil-core/src/nonAndroidMain/kotlin/coil3/CoilPainter.kt

```diff
@@ -0,0 +1,18 @@
+package coil3
+
+import coil3.annotation.ExperimentalCoilApi
+import org.jetbrains.skia.Bitmap
+import org.jetbrains.skia.Canvas
+
+@ExperimentalCoilApi
+sealed interface CoilPainter {
```

Comment:
I think this API isn't the right abstraction as it locks us into a fixed set of supported use cases via the `sealed interface`. For example, how would we add future support for rendering an animated gif without adding another sealed type?

It also feels like we're duplicating functionality from the `Image` class here. I think we should remove this type and add `fun Canvas.onDraw()` to the common `Image` type.

Reply:
But then how to decide how the `Image` is going to be converted to a Compose `Painter`.

Reply:
I think ideally we can create an `ImagePainter` which wraps a generic `Image` and draws it

Reply:
The idea of `ImagePainter` is not clear to me.
Check the new changes, I removed the `CoilPainter` and added `fun Canvas.onDraw()` to `Image`.